crates/re_sdk/src/log_sink.rs

```diff
@@ -148,11 +157,19 @@ impl LogSink for MemorySink {
     #[inline]
     fn send(&self, msg: LogMsg) {
         self.0.write().push(msg);
+
+        if let Some(flush_hook) = self.0.flush_hook.as_ref() {
```

Comment:
The problem I had was caused by trying to call the hook in `flush_blocking` instead of here. At first I misunderstood what `LogSink::send` does, I thought it was called for each `rr.log`, but it seems to be called for a single _batch_ of messages, all encoded as one `ArrowMsg`. So it seems fine to put this here, if it's not used then it gets branch predicted to oblivion